src/useCustomFetchSuspense.ts

```diff
@@ -0,0 +1,30 @@
+import produce from 'immer';
+import lru from 'lru-cache';
+import md5 from 'md5';
+
+const fetchData = (url: RequestInfo, options: RequestInit | undefined = {}) => {
+    return fetch(url, options).then(res => res.json());
+}
+
+const cache = new lru(50);
+
+export const useCustomFetchSuspense = (url: RequestInfo, options: RequestInit | undefined = {}) => {
+    const key = `${url}.${md5(JSON.stringify(options))}`;
+
+    const value: any = cache.get(key) || { status: 'new', data: null }
+
+    if(value.status === 'resolved'){
+        return value.data;
+    }
+
+    const promise = fetchData(url, options);
+
+    promise.then(data => {
```

Comment:
Couldnt a pending, non-failed, non-resolved promise be put into the cache so that multiple duplicate requests are deduplicated?